mne/time_frequency/tfr.py

```diff
@@ -100,6 +100,58 @@ def morlet(sfreq, freqs, n_cycles=7.0, sigma=None, zero_mean=False):
     return Ws
 
 
+def wavelet(sfreq, freqs, n_cycles=7.0, window="hann"):
+    """Compute wavelets for the given frequency range.
```

Comment:
what type of wavelet are you implementing here? we have morlet which is one type of wavelet. You have others like Morse etc...  There should be one way to building the different wavelet functions with a consitent API.

Reply:
I was looking for a window to apply in my tfr analysis something like what fieldtrip uses. I didn't find anything similar to that functionality, therefore I decided to add the window with the same oscillatory signal used in Morlet. Perhaps, it Is better to change the name of the function

Reply:
I know that in neuroscience we tend to call an oscillatory signal a wavelet but to me it's really not good practice. A wavelet should be zero mean and of unit norm see: https://en.wikipedia.org/wiki/Wavelet

here you use Fourier atom with a window. How is it called in Fieldtrip?

one of problem if your window is not zero mean is that if you have a constant signal you will get a non-zero TFR.

Reply:
> here you use Fourier atom with a window. How is it called in Fieldtrip?

And in one special case -- when the window is Gaussian -- it's a Morlet wavelet. I don't know if there are other wavelets that are just a complex exponential (Fourier atom) multiplied by a window. But calling this function `wavelet` indeed seems misleading...

Reply:
The fieldtrip uses 
function [freq] = ft_freqanalysis(cfg, data)
where cfg is a struct defining how to implement the analysis. When you check the code for its hanning window it just gets the signal multiplied by the Fourier atom and apply that window by multiplication.

Reply:
> I know that in neuroscience we tend to call an oscillatory signal a wavelet but to me it's really not good practice. A wavelet should be zero mean and of unit norm see: https://en.wikipedia.org/wiki/Wavelet
> 
> here you use Fourier atom with a window. How is it called in Fieldtrip?
> 
> one of problem if your window is not zero mean is that if you have a constant signal you will get a non-zero TFR.

The fieldtrip uses
function [freq] = ft_freqanalysis(cfg, data)
where cfg is a struct defining how to implement the analysis. When you check the code for its hanning window it just gets the signal multiplied by the Fourier atom and apply that window by multiplication.